gdal/port/cpl_http.cpp

```diff
@@ -583,20 +633,41 @@ CPLHTTPResult *CPLHTTPFetch( const char *pszURL, char **papszOptions )
     curl_easy_setopt(http_handle, CURLOPT_HEADERFUNCTION, CPLHdrWriteFct);
 
     CPLHTTPResultWithLimit sResultWithLimit;
-    sResultWithLimit.psResult = psResult;
-    sResultWithLimit.nMaxFileSize = 0;
-    const char* pszMaxFileSize = CSLFetchNameValue(papszOptions,
-                                                   "MAX_FILE_SIZE");
-    if( pszMaxFileSize != nullptr )
+    if( nullptr == pfnWrite )
     {
-        sResultWithLimit.nMaxFileSize = atoi(pszMaxFileSize);
-        // Only useful if size is returned by server before actual download.
-        curl_easy_setopt(http_handle, CURLOPT_MAXFILESIZE,
-                         sResultWithLimit.nMaxFileSize);
+        // Fix error C2440: This conversion requires a reinterpret_cast, a C-style 
+        // cast or function-style cast.
+        pfnWrite = reinterpret_cast<CPLHTTPFetchWriteFunc>(CPLWriteFct);
```

Comment:
This cast should hopefully be removed if removing the CPL_STDCALL from CPLHTTPFetchWriteFunc